src/hotspot/cpu/x86/continuationFreezeThaw_x86.inline.hpp

```diff
@@ -136,7 +142,9 @@ inline void FreezeBase::relativize_interpreted_frame_metadata(const frame& f, co
   assert((intptr_t*)hf.at_relative(frame::interpreter_frame_last_sp_offset) == hf.unextended_sp(), "");
 
   // Make sure that locals is already relativized.
-  assert((*hf.addr_at(frame::interpreter_frame_locals_offset) == frame::sender_sp_offset + f.interpreter_frame_method()->max_locals() - 1), "");
+  DEBUG_ONLY(Method* m = f.interpreter_frame_method();)
+  DEBUG_ONLY(int max_locals = !m->is_native() ? m->max_locals() : m->size_of_parameters() + 2;)
```

Comment:
What is the + 2 for?  Is the check for is_native because of wait0?  Please add a comment what this line is doing.

Reply:
It's for the 2 extra words for native methods (temp oop/result handler). Added comment.